osquery/logger/plugins/aws_kinesis.cpp

```diff
@@ -65,41 +65,69 @@ Status KinesisLoggerPlugin::logString(const std::string& s) {
 
 Status KinesisLogForwarder::send(std::vector<std::string>& log_data,
                                  const std::string& log_type) {
-  std::vector<Aws::Kinesis::Model::PutRecordsRequestEntry> entries;
-  if (FLAGS_aws_kinesis_random_partition_key) {
-    boost::uuids::uuid uuid = boost::uuids::random_generator()();
-    partition_key_ = boost::uuids::to_string(uuid);
-  }
-  for (const std::string& log : log_data) {
-    if (log.size() > kKinesisMaxLogBytes) {
-      LOG(ERROR) << "Kinesis log too big, discarding!";
+  int retry_count = 100;
+  int retry_delay = 3000;
+  std::vector<std::string> batch = log_data;
```

Comment:
This is potentially VERY expensive. `log_data` is passed to `::send` as a mutable reference, can we use that instead of making a copy?